### PR TITLE
Containerize the box

### PR DESCRIPTION
**Pull request by scopatz**
This PR let's xonsh run inabox in a container from the browser. The auto-encryption stuff doesn't work for me yet so it is commented out.  Also, since this pulls from this repo, some of the URLs that point to this branch will have to be renamed to master once this is merged.

### CODE REVIEW

File: install [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
 
-sudo useradd xonsh
-
-sudo cp $BASE/pam.conf /etc/pam.d/shellinabox
-sudo cp $BASE/upstart.conf /etc/init/shellinabox.conf
-sudo cp $BASE/xonshrc ~xonsh/.xonshrc
+# Some user setup
+useradd xonsh
+cp $BASE/pam.conf /etc/pam.d/shellinabox
+cp $BASE/upstart.conf /etc/systemd/system/shellinabox.service
```
Comment (AstraLuma) [inline install]: I'm about 95% certain this is NOT how that works.

Comment (scopatz) [reply install]: Yeah, I don't think that line is needed even, since we aren't relying on systemd to run the command. I have removed it.

**Comment by scopatz** [comment]
Let me know if you think it is ready!

**Comment by AstraLuma** [comment]
What starts `shellinaboxd`?

**Comment by scopatz** [comment]
It seems like you can pass in a command to run via the AWS docker service.  Maybe that isn't robust enough?

**Comment by AstraLuma** [comment]
Can you turn the command that was in the upstart config into a script to be executed? About 90% of the runtime configuration is in that command.

Oh, and double check the flags? Some of them might be things like daemonization.

**Comment by scopatz** [comment]
Ok I think I have it working now. You can set the run command in the dockerfile

File: systemd.conf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# shellinabox.service
```
Comment (AstraLuma) [inline systemd.conf]: Why do we need the systemd file if the entry point is defined in the docker file?

Comment (scopatz) [reply systemd.conf]: We don't, I can remove all of that and the upstart stuff, if you want.

Comment (scopatz) [reply systemd.conf]: Just let me know

Comment (AstraLuma) [reply systemd.conf]: Lets not leave redundant stuff around like this. It's just going to get dusty and outdated.

Comment (scopatz) [reply systemd.conf]: Ok cleaned up

**Comment by scopatz** [comment]
I think this is ready now.

**Pull request merged**